library/BerlinPipeline/src/Juvix/BerlinPipeline/Env.hs

```diff
@@ -37,23 +39,65 @@ newtype EnvS b = EnvS (State T b)
 
 data StopADT = Stop
 
+-- | Register the pipeline function to the environment
 registerStep :: CircularList.T Step.Named -> EnvS ()
-registerStep = notImplemented
+registerStep l = do
+  modify@"registeredPipeline" $ \i -> i <> l
 
+-- | Create a named group of pipeline steps or nested grouping of pipeline steps.
 defPipelineGroup :: NameSymbol.T -> [CircularList.T Step.Named] -> CircularList.T Step.Named
-defPipelineGroup = notImplemented
+defPipelineGroup sym ls = foldl' (<>) (CircularList.init sym) ls
```

Comment:
Maybe we should use `CircularList.empty` instead here?

We should probably change the format of these to have name be had over groups of passes, as currently this is unable to actually add the name